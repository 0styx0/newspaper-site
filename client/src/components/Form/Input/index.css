input:not([type=number]):not([type=submit]):not([type=checkbox]) {
    height: 16px;
    width: 144px;
}