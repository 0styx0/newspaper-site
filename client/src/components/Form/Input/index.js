import React from 'react';
import Label from '../Label';

import './index.css';

/**
 * @prop label, required, name, children - @see Select
 * @prop abbr - optional text for abbr elt if want to give user a hint about the input
 * @prop props - json of any native attr wanted
 *
 * @return input elt wrapped in a @see Label
 */

export default function Input(props) {

    const input = React.cloneElement(<input />, props.props);

    return (
        <Label
        value={props.label}
        required={!!props.props.required}
        abbr={props.abbr}

        children={
            input
        }

        />
    );
}

