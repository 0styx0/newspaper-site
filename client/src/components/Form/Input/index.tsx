import * as React from 'react';
import Label from '../Label';
import './index.css';

interface Props {
    label: string;
    abbr?: string;
    // I can't find a way to just say that html attributes should be accepted
    props: Object;
}

/**
 * @prop label, required, name, children - @see Select
 * @prop abbr - optional text for abbr elt if want to give user a hint about the input
 * @prop props - json of any native attr wanted
 *
 * @return input elt wrapped in a @see Label
 */
export default function Input(props: Props) {

    const input = React.cloneElement(<input />, props.props);

    return (
        <Label
            value={props.label}
            required={!!props.props.required}
            abbr={props.abbr}
            children={
                input
            }
        />
    );
}
