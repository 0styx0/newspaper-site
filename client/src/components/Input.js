import React from 'react';

/**
 * @prop label, children, abbr, required @see Select
 *
 * @return html label as parent to props.children with text of prop.value concatted with ': ',
   if abbr is given, that's put in too, and if required is given, a red asterisk is put in too
 */
function Label(props) {

    const label = props.value + ": ";

    return (
        <label>{label}
            {props.children}
            {!!props.abbr ? <abbr title={props.abbr}>?</abbr> : ""}
            {!!props.required ? <span className="danger">*</span> : ""}
        </label>
    )
}

/**
 * @prop label, required, name, children - @see Select
 * @prop abbr - optional text for abbr elt if want to give user a hint about the input
 * @prop placeholder = html attr
 * @prop type = html attr
 * @prop title = html attr
 * @prop pattern = html attr
 *
 * @return input elt wrapped in a @see Label
 */
function Input(props) {


    return (
        <Label
          value={props.label}
          required={!!props.required}
          abbr={props.abbr}

          children={

            <input
              name={props.name}
              type={props.type}
              placeholder={props.placeholder}
              pattern={props.pattern}
              required={!!props.required}
              title={props.title}
            />}
        />
    );
}

/**
 * @prop label - string, contents of label element
 * @prop required - if it exists, the select gets `required attribute
 * @prop name - name of select
 * @prop children - option elements
 */
function Select(props) {

    return (
        <Label
            value={props.label}
            required={!!props.required}

            children={
                <select name={props.name} onChange={props.onChange}>
                    {props.children}
                </select>
            }
        />
    );
}

export {
    Input,
    Select
};