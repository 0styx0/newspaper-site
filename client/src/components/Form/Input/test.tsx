import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Input from './';

describe('<Input>', () => {

    describe('snapshots', () => {

        /**
         * Takes snapshots
         */
        function snap(abbr?: string) {

            const tree = renderer.create(
                <Input
                    label="Random"
                    props={{
                        placeholder: 'Hi'
                    }}
                    abbr={abbr}
                />
            ).toJSON();

            expect(tree).toMatchSnapshot();

        }

        it('renders correctly when abbr is given', () => snap('A very good hint'));

        it('renders correctly when abbr is NOT given', () => snap());
    });
});
