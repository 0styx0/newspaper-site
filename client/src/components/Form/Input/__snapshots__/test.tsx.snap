// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> snapshots renders correctly when abbr is NOT given 1`] = `
<label>
  Random: 
  <input
    placeholder="Hi"
  />
  
</label>
`;

exports[`<Input> snapshots renders correctly when abbr is given 1`] = `
<label
  onInput={[Function]}
>
  Random: 
  <input
    placeholder="Hi"
  />
  
  <span>
    <abbr
      onClick={[Function]}
      title="A very good hint"
    >
      ?
    </abbr>
    <span />
  </span>
</label>
`;
